### PR TITLE
Added BasisFunctionTests.cpp. Removed BasisFunction unit test and tru…

### PR DESCRIPTION
**Pull request by paarongiroux**
…thfile.

## Description

I created gtests for the BasisFunction.cpp and deleted the BasisFunction unit test and truth file.
## Motivation and Context

Part of improving ISIS3 tests
## How Has This Been Tested?

This is a test.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by jessemapel** [comment]
Needs the PR template filled out

File: isis/tests/BasisFunctionTests.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+  EXPECT_EQ(testBasis.Coefficients(), 3);
+  EXPECT_EQ(testBasis.Variables(), 3);
+  EXPECT_STREQ(testBasis.Name().toStdString().c_str(), name.c_str());
+}
+
+TEST(BasisFunction, Evaluation)
+{
+  Isis::BasisFunction testBasis("basis", 3, 3);
+  double expectedOutput = 2.5*3.0 + 3.2 + 2.0;
```
Comment (Kelvinrr) [inline isis/tests/BasisFunctionTests.cpp]: Why not just have the result here (12.7 in this case)?

File: isis/tests/BasisFunctionTests.cpp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  vars.push_back(2.0);
+
+  output = testBasis.Evaluate(vars);
+  EXPECT_DOUBLE_EQ(output, expectedOutput);
+
+  vars[0] = 3.5;
+  vars[1] = 1.2;
+  vars[2] = 10.8;
+
+  expectedOutput = 2.5 * 3.5 + 3.2 * 1.2 + 10.8;
```
Comment (Kelvinrr) [inline isis/tests/BasisFunctionTests.cpp]: same here

File: isis/tests/BasisFunctionTests.cpp [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+               "The number of terms in the expansion does not "
+               "match number of coefficients in the basis equation\"";
+  }
+}
+
+TEST(BasisFunction, NonVectorVariable)
+{
+  Isis::BasisFunction testBasis("basis", 1, 1);
+  double var = 1.2;
+  double expectedOutput = 1.2 * 2.5;
```
Comment (Kelvinrr) [inline isis/tests/BasisFunctionTests.cpp]: and here

File: isis/tests/BasisFunctionTests.cpp (hunk 1 above)
Comment (paarongiroux) [reply isis/tests/BasisFunctionTests.cpp]: I just did that out of laziness. I just changed them to the simplified versions so everything should be good now.

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**